web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -0,0 +1,401 @@
+    var result = (a > b ? 1 : (a < b ? -1 : 0));
+    return (result || Utils.sortById(aRaw, bRaw)) * sortDirection;
+  }
+
+  searchData(datum, criteria) {
+      if (criteria) {
+        return datum.name.toLowerCase().includes(criteria.toLowerCase());
+      }
+      return true;
+  }
+
+  buildRows(guests) {
+    return Object.keys(guests).map((id) => guests[id]);
+  }
+
+  render() {
+    return (<VirtualizationGuestActionApi hostid={this.props.server_id}
+                                          callback={this.actionCallback}>
+      {
+        ({
+          onAction,
+          messages,
+        }) => {
+          const data = this.state.serverData ? this.state.serverData : [];
+          const modals_data = [
+            {'type': 'start', 'name': t('Start / Resume'), 'icon': 'fa-play', 'bulkonly': true},
+            {'type': 'shutdown', 'name': t('Stop'), 'icon': 'fa-stop', 'bulkonly': false},
+            {'type': 'restart', 'name': t('Restart'), 'icon': 'fa-refresh', 'bulkonly': false},
+            {'type': 'suspend', 'name': t('Suspend'), 'icon': 'fa-pause', 'bulkonly': false},
+            {'type': 'delete', 'name': t('Delete'), 'icon': 'fa-trash', 'bulkonly': false}
+          ];
+          const panelButtons =
+            <div className="pull-right btn-group">
+              {modals_data
+                .filter(action => action.type != 'delete' || this.props.salt_entitled)
+                .map(action => this.createSelectedModalButton(action))}
+            </div>;
+
+          return (
+            <VirtualizationGuestsListRefreshApi server_id={this.props.server_id} refreshInterval={this.props.refreshInterval}>
+            {
+              ({
+                guests,
+                refreshError,
+              }) => {
+                return (
+                  <div>
+                    {panelButtons}
+                    <h2>{t("Hosted Virtual Systems")}</h2>
+                    <p>{t("This is a list of virtual guests which are configured to run on this host.")}</p>
+                    <MessageContainer items={this.state.errors.map(msg => MessagesUtils.error(msg))}/>
+                    { refreshError && <MessageContainer items={refreshError}/> }
+                    <Table
+                      data={this.buildRows(guests)}
+                      emptyText={t('No virtual guest to show.')}
+                      identifier={(row) => row.uuid}
+                      initialSortColumnKey="name"
+                      initialItemsPerPage={userPrefPageSize}
+                      selectable
+                      selectedItems={this.state.selectedItems}
+                      onSelect={this.handleSelectItems}>
+                      searchField={
+                          <SearchField filter={this.searchData}
+                              criteria={""}
+                              placeholder={t("Filter by name")} />
+                      }>
+                      <Column
```

Comment:
Point for future:

I wonder if this logic couldn't be simplified in a more declarative way, it seems way hard to read and change.

```javascript
const columns = [
   { key="serverName", comparator=Utils.sortByText, etc}
]
columns.push(generateColumnStatus())
columns.push(generateColumnType())

return (
   columns.map((column) => <column ...columns />)
)
```

Reply:
then that should be applied to all tables: there is surely something to do on the component side